### ISSUE TITLE
reproduce a score

### ISSUE THREAD

**Issue by chiyuan1126** [BaderLab/Saber#197]
Hello @JohnGiorgi , Thanks very much for sharing such a great work!! 

When I was trying to reproduce a score of your paper. “END-TO-END NAMED ENTITY RECOGNITION AND RE- LATION EXTRACTION USING PRE-TRAINED LANGUAGE MODELS ” , the NER_loss and RE_loss did not go down. 

I was testing it with the latest development branch on ADE corpus (I generated the testing set and used the training set you provide in bowang-lab/joint-ner-and-re repo.) . I used biobert_v1.1_pubmed  as the pre-trained model. I trained entity-relation jointly extraction model multiple epochs but the loss did not go down and the evaluation results seems not very good. I guess it might be caused by my incorrect configuration or I might not setup Saber correctly. 

I added below code for reproducing joint-ner-and-re
` 

        def train_joint_ner_rel(self, directory=None, type=None):
                self.config.dataset_reader="conll2004datasetreader"
                self.config.verbose= True
                if directory is not None:
                    dataset_folder = directory if isinstance(directory, list) else [directory]
                    dataset_folder = [generic_utils.clean_path(dir_) for dir_ in dataset_folder]
                    self.config.dataset_folder = dataset_folder

                if not self.config.dataset_folder:
                    err_msg = ('Expected a valid path in "directory" or "self.config.dataset_folder".'
                               f' Got {directory} and {self.config.dataset_folder} respectively')
                    LOGGER.error('ValueError %s', err_msg)
                    raise ValueError(err_msg)

                # Datasets may be 'single' or 'compound' (more than one)
                if len(self.config.dataset_folder) == 1:
                    print('Loading (single) dataset...', end=' ', flush=True)
                    self.datasets = data_utils.load_single_dataset(self.config)
                    #print(self.datasets._get_type_seq())
                from .models.bert_for_ner_and_re import BertForNERAndRE
                start = time.time()
                if not self.datasets:
                    err_msg = ('`Saber.datasets` is empty. Make sure to load at least one dataset with'
                               ' `Saber.load_dataset()` before calling `Saber.build()`.')
                    LOGGER.error('MissingStepException: %s', err_msg)
                    raise MissingStepException(err_msg)
       
                print('Building the BERT model for joint NER and RE...', end=' ', flush=True)
            
                model = BertForNERAndRE(config=self.config,datasets=self.datasets, pretrained_model_name_or_path=constants.PYTORCH_BERT_MODEL)
                model.config.epochs = 1
                model.specify()
                self.models.append(model)
                print(f'Done ({time.time() - start:.2f} seconds).')
                LOGGER.info('%s model was built successfully', self.config.model_name.upper())
                self.models[-1].train();`

Any suggestions or comments would be greatly appreciated!!

**Comment by chiyuan1126** [comment]
Sorry, I did not notice the default learning rate is 0.  😂😂😂  After I set up the learning rate according to the paper,  it works perfect now!  Thanks again for sharing this great tool. It's fantastic!

**Comment by JohnGiorgi** [comment]
Okay awesome, good catch. Thanks for letting me know!
